src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        public static void Initialize()
+        {
+            Initialize(errors: null, modulePathList: null);
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">List of module path from where DSC PS modules will be loaded.</param>
+        public static void Initialize(Collection<Exception> errors, List<string> modulePathList)
+        {
+            s_tracer.WriteLine("Initializing DSC class cache");
+
+            if (Platform.IsLinux || Platform.IsMacOS)
+            {
+                // Load the base schema files.
+                ClearCache();
+                var dscConfigurationDirectory = Environment.GetEnvironmentVariable("DSC_HOME");
+                if (string.IsNullOrEmpty(dscConfigurationDirectory))
+                {
+                    // if DSC_HOME env var is not set, then use location of system-wide PS module directory (i.e. /usr/local/share/powershell/Modules) as backup
+                    dscConfigurationDirectory = Path.Join(ModuleIntrinsics.GetSharedModulePath(), "PSDesiredStateConfiguration", "Configuration");
+                }
+
+                if (!Directory.Exists(dscConfigurationDirectory))
+                {
+                    throw new DirectoryNotFoundException(string.Format(ParserStrings.PsDscMissingSchemaStore, dscConfigurationDirectory));
+                }
+
+                var resourceBaseFile = Path.Join(dscConfigurationDirectory, "BaseRegistration", "BaseResource.schema.json");
+                ImportClasses(resourceBaseFile, s_defaultModuleInfoForResource, errors);
+                var metaConfigFile = Path.Join(dscConfigurationDirectory, "BaseRegistration", "MSFT_DSCMetaConfiguration.json");
+                ImportClasses(metaConfigFile, s_defaultModuleInfoForResource, errors);
+
+                var allResourceRoots = new string[] { dscConfigurationDirectory };
+
+                // Load all of the system resource schema files, searching
+                string resources;
+                foreach (var resourceRoot in allResourceRoots)
+                {
+                    resources = Path.Join(resourceRoot, "schema");
+                    if (!Directory.Exists(resources))
+                    {
+                        continue;
+                    }
+
+                    foreach (var schemaFile in Directory.EnumerateFiles(resources, "*.schema.json", SearchOption.AllDirectories))
+                    {
+                        ImportClasses(schemaFile, s_defaultModuleInfoForResource, errors);
+                    }
+                }
+            }
+            else
```

Comment:
This seems like a good place to use `return` rather than `else`. Really this is two methods, one for Windows and the other for Unix

Reply:
For ease of cross-referencing with existing mof-version I'd like to leave this as is.